### PR TITLE
Fold `verify-all-modules-with-headSparkVersion` into `verify-all-modules` [skip ci]

### LINKED ISSUE #8401: Fold `verify-all-modules-with-headSparkVersion` into  `verify-modules-with-jdk`
verify-all-modules-with-headSparkVersion can be subsumed by a more general matrix job verify-all-modules-with-jdk using 8 as jdk parameter.
      
_Originally posted by @gerashegalov in https://github.com/NVIDIA/spark-rapids/issues/8360#issuecomment-1560349338_

### PR DESCRIPTION
**Pull request by YanxuanLiu**
fix #8401 

`verify-all-modules-with-headSparkVersion` has exactly same steps with `verify-all-modules`. So we add the combination of `jdk8` and `headSparkVersion` into `verify-all-modules` matrix. 

Also we renamed `verify-modules-with-jdk` to `verify-all-modules` to avoid misunderstanding, since the test cares about both jdk version and spark version.

### CODE REVIEW

**Comment by YanxuanLiu** [comment]
The jdk step inputs jdk version into `mvn -P`, like `'individual,pre-merge,jdk11'`, but head version step does not, `'individual,pre-merge'`.

In this case, I make activate profiles as an item of matrix. https://github.com/NVIDIA/spark-rapids/pull/8564/files#diff-031910b9e60925c8a39b96ae87fd04572bc53b41b28ced0fbb35b95790cf9613R67

Can we align the arguments of `mvn -P` for these 2 steps, so that we don't need to add profiles to the matrix. cc @pxLi

**Comment by YanxuanLiu** [comment]
> 



> > The jdk step inputs jdk version into `mvn -P`, like `'individual,pre-merge,jdk11'`, but head version step does not, `'individual,pre-merge'`.
> > In this case, I make activate profiles as an item of matrix. https://github.com/NVIDIA/spark-rapids/pull/8564/files#diff-031910b9e60925c8a39b96ae87fd04572bc53b41b28ced0fbb35b95790cf9613R67
> > Can we align the arguments of `mvn -P` for these 2 steps, so that we don't need to add profiles to the matrix. cc @pxLi
> 
> sound OK to me. Also please remove/comment out verify-all-modules-with-headSparkVersion in this PR even its still in draft to make it clear whats the purpose for the change, thx

OK

Removed profiles from matrix. Just added jdk8 and head spark version as default version to the matrix.

Also renamed the step as `verify-all-modules`, the original name `verify-modules-with-jdk` may cause misunderstanding that the step is only a JDK test.

**Comment by tgravescs** [comment]
can we please have a more descriptive title and description on this?   It should be descriptive about what and why the change is being made.

File: .github/workflows/mvn-verify-check.yml [hunk 1]
```diff
@@ -99,28 +101,7 @@ jobs:
...
-      # includes RAT, code style and doc-gen checks of default shim
-      - name: verify all modules with lowest-supported Spark version
-        run: >
-          mvn -Dmaven.wagon.http.retryHandler.count=3 -B verify
-          -P 'individual,pre-merge'
-          -Dbuildver=${{ needs.get-shim-versions-from-dist.outputs.sparkHeadVersion }}
-          $COMMON_MVN_FLAGS
-
-  verify-modules-with-jdk:
+  verify-all-modules:
```
[the remark this answers is not in the document]
Comment (YanxuanLiu) [reply .github/workflows/mvn-verify-check.yml:102]: removed `sparkHeadVersion` since we can get the same value from `SPARK_BASE_SHIM_VERSION` directly.

**Review by gerashegalov — APPROVED** [review]

**Comment by gerashegalov** [comment]
build

**Review by pxLi — APPROVED** [review]

**Pull request merged**